File: 05-01-2025/findDisappearedNums.js
function findDisappearedNums(nums) {
  const num_set = new Set(nums);
  let len = nums.length;
  let result = [];

  for (i = 1; i < len + 1; i++) {
    if (!num_set.has(i)) {
      result.push(i);
    }
  }

  return result;
}

console.log(findDisappearedNums([4, 3, 2, 7, 8, 2, 3, 1]));

File: 04-01-2025/findDisappearedNums.py
def findDisappearedNumbers(nums):
    result = []
    num_set = set(nums)  # Set is created once
    l = len(nums)

    for i in range(1, l + 1):
        if i not in num_set:  # Membership check is efficient (O(1))
            result.append(i)

    return result

# avoid creating sets instide loops